lib/modules/manager/composer/extract.ts

```diff
@@ -130,9 +130,7 @@ export async function extractPackageFile(
               depType,
               depName,
               currentValue,
-              datasource: GithubTagsDatasource.id,
-              packageName: 'php/php-src',
-              extractVersion: '^php-(?<version>.*)$',
+              datasource: DockerDatasource.id,
```

Comment:
i think we also need extract version and versioning, as the default docker versioning applied by datasource won't work as expected.

Reply:
And looking through the git tags datasource, I don't even see a `defaultVersioning` set, that's why I assumed it must be picking that up from composer instead. For docker it does specify one
https://github.com/renovatebot/renovate/blob/c56588eaba967cd6b4ba103b8b30c0dc52b3addf/lib/modules/datasource/docker/index.ts#L379

Would that be enough for docker versioning to be used or do I still need to specify it here?

Reply:
docker versioning won't update from 1.2.3 to 1.3 or 1.2.3-rc1.

docker versioning treats the suffix as compatibility value which needs to be equal on all versions.

that because we don't want to update from 1.2.3 to 1.2.4-alpine or something like that.

also the version parts need to be on same length.

so that's why I suggest to add extract version and versioning.

use a very specific regex to select versions, which doesn't cause false positives.

Reply:
I'm not sure what those values should be though. Would you expect something along the lines of
```ts
datasource: DockerDatasource.id,
extractVersion: '^(?<version>[^-]*)`,
versioning: phpVersioning.id,
```

where `phpVersioning.id` comes from a new custom versioning scheme which can convert `8.2.0RC5` to `8.2.0-rc5`?

Reply:
no, the datasource needs to return right values. if it's not possible to get that version from an existing datasource with extract version, then we need a custom php datasource, which returns the right versions.

Reply:
Ok, thanks! So if I create something like https://github.com/renovatebot/renovate/blob/main/lib/modules/datasource/node/index.ts but for PHP then?

I will push those changes to this PR, or if you want to close this, I'll reopen another one with those changes. Probably tomorrow now though. Thanks for your help with this!

Reply:
Many things were discussed in https://github.com/renovatebot/renovate/discussions/18695, but this PR has now been updated to address those from there and these here.